src/OpenCV_Mat.jl

```diff
@@ -1,225 +1,77 @@
 {
-         for (int i = 0; i < 3; i++) { img.at<cv::Vec3f>(row,col)[i] = static_cast<float>(vec[i]); };
+    std::vector<T1>vec(3);
+    for(int i = 0; i < 3; ++i)
+    {
+        vec[i] = img.at<cv::Vec<T1, 3>>(row, col)[i];
+    }
+    return vec;
 }
 
-std::vector<double> at_v3d(cv::Mat &img, int row, int col)
+template <typename T2>
+inline void ptr_val3(cv::Mat &img, int row, int col, std::vector<double> vec)
 {
-         std::vector<double>vec(3);
-         for (int i = 0; i < 3; i++) { vec[i]= img.at<cv::Vec3d>(row,col)[i]; }; return vec;
+    for(int i = 0; i < 3; ++i)
+    {
+        img.at<cv::Vec<T2, 3>>(row, col)[i] = static_cast<T2>(vec[i]);
+    }
 }
 
-void at_v3d(cv::Mat &img, int row, int col, std::vector<double> vec)
-{
-         for (int i = 0; i < 3; i++) { img.at<cv::Vec3d>(row,col)[i] = vec[i]; };
-}
 """
 
 function pixget(img, row::Int, col::Int)
   (row < 0 || col < 0 || row > rows(img) || col > cols(img)) ? throw(BoundsError()) : nothing
 
-  # Grayscale and binary images (returns int)
-  if (cvtypeval(img) == CV_8UC1)
-    val = @cxx at_u(img, row, col);
-    #println(int(val)); return (val)
-  elseif (cvtypeval(img) == CV_16SC1)
-    val = @cxx at_s(img, row, col);
-    println(int(val)); return (val)
-  elseif (cvtypeval(img) == CV_32SC1)
-    val = @cxx at_us(img, row, col);
-    println(int(val)); return (val)
-  elseif (cvtypeval(img) == CV_32FC1)
-    val = @cxx at_f(img, row, col);
-    println(val); return (val)
-  elseif (cvtypeval(img) == CV_64FC1)
-    val = @cxx at_d(img, row, col);
-    println(val); return (val)
-  # RGB images (returns vec with original types)
-  elseif (cvtypeval(img) == CV_8SC3)
-    vec = @cxx at_vc(img, row, col);
-    println([int(at(vec, 0)), int(at(vec, 1)),int(at(vec, 2))]); return (vec)
-  elseif (cvtypeval(img) == CV_8UC3)
-    vec = @cxx at_v3b(img, row, col);
-    println([int(at(vec, 0)), int(at(vec, 1)),int(at(vec, 2))]); return (vec)
-  elseif (cvtypeval(img) == CV_16SC3)
-    vec = @cxx at_v3s(img, row, col);
-    println([int(at(vec, 0)), int(at(vec, 1)),int(at(vec, 2))]); return (vec)
-  elseif (cvtypeval(img) == CV_16UC3)
-    vec = @cxx at_v3us(img, row, col);
-    println([int(at(vec, 0)), int(at(vec, 1)),int(at(vec, 2))]); return (vec)
-  elseif (cvtypeval(img) == CV_32SC3)
-    vec = @cxx at_v3i(img, row, col);
-    println([int(at(vec, 0)), int(at(vec, 1)),int(at(vec, 2))]); return (vec)
-  elseif (cvtypeval(img) == CV_32FC3)
-    vec = @cxx at_v3f(img, row, col);
-    println([float(at(vec, 0)), float(at(vec, 1)),float(at(vec, 2))]); return (vec)
-  elseif (cvtypeval(img) == CV_64FC3)
-    vec = @cxx at_v3d(img, row, col);
-    println([at(vec, 0), at(vec, 1),at(vec, 2)]); return (vec)
-  else throw(ArgumentError("Image format not recognized!"))
-  end
+  # Grayscale and binary images
+  val = @cxx ptr_val(img, row, col);
+  println(val)
+  return (val)
+
+  # RGB images
+  vec = @cxx ptr_val3(img, row, col);
```

Comment:
This statement will not be reached because there is no conditional statement anymore in the function.